ui/src/app/pages/data/data-set/new-dataset-modal.tsx

```diff
@@ -325,8 +331,17 @@ class NewDataSetModal extends React.Component<Props, State> {
           Cancel
         </Button>
         <TooltipTrigger content={summarizeErrors(errors)}>
-          <Button type='primary' data-test-id='save-data-set'
-                  disabled={errors} onClick={() => this.saveDataSet()}>
+          <Button type='primary'
+                  data-test-id='save-data-set'
+                  disabled={errors}
+                  onClick={() => {
+                    if (this.props.dataSet) {
```

Comment:
I am not sure if its good to have if else in render, it would be better to have it as a function